RedfishPkg/RedfishCredentialDxe/RedfishCredentialDxe.h

```diff
@@ -10,68 +11,49 @@
 
-#include <Protocol/EdkIIRedfishCredential.h>
+#include <Protocol/EdkIIRedfishCredential2.h>
 
 #include <Library/BaseLib.h>
 #include <Library/DebugLib.h>
 #include <Library/PrintLib.h>
 #include <Library/RedfishCredentialLib.h>
 #include <Library/UefiLib.h>
 #include <Library/UefiBootServicesTableLib.h>
+#include <Library/RedfishHttpLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/RedfishDebugLib.h>
 
-/**
-  Retrieve platform's Redfish authentication information.
-
-  This functions returns the Redfish authentication method together with the user Id and
-  password.
-  - For AuthMethodNone, the UserId and Password could be used for HTTP header authentication
-    as defined by RFC7235.
-  - For AuthMethodRedfishSession, the UserId and Password could be used for Redfish
-    session login as defined by  Redfish API specification (DSP0266).
-
-  Callers are responsible for and freeing the returned string storage.
-
-  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL_PROTOCOL instance.
-  @param[out]  AuthMethod          Type of Redfish authentication method.
-  @param[out]  UserId              The pointer to store the returned UserId string.
-  @param[out]  Password            The pointer to store the returned Password string.
-
-  @retval EFI_SUCCESS              Get the authentication information successfully.
-  @retval EFI_ACCESS_DENIED        SecureBoot is disabled after EndOfDxe.
-  @retval EFI_INVALID_PARAMETER    This or AuthMethod or UserId or Password is NULL.
-  @retval EFI_OUT_OF_RESOURCES     There are not enough memory resources.
-  @retval EFI_UNSUPPORTED          Unsupported authentication method is found.
-
-**/
-EFI_STATUS
-EFIAPI
-RedfishCredentialGetAuthInfo (
-  IN  EDKII_REDFISH_CREDENTIAL_PROTOCOL  *This,
-  OUT EDKII_REDFISH_AUTH_METHOD          *AuthMethod,
-  OUT CHAR8                              **UserId,
-  OUT CHAR8                              **Password
-  );
-
-/**
-  Notify the Redfish service provide to stop provide configuration service to this platform.
-
-  This function should be called when the platfrom is about to leave the safe environment.
-  It will notify the Redfish service provider to abort all logined session, and prohibit
-  further login with original auth info. GetAuthInfo() will return EFI_UNSUPPORTED once this
-  function is returned.
-
-  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL_PROTOCOL instance.
+#define REDFISH_CREDENTIAL_VERSION              0x00010000
+#define REDFISH_CREDENTIAL_DEBUG                DEBUG_VERBOSE
+#define REDFISH_MANAGER_ACCOUNT_COLLECTION_URI  L"AccountService/Accounts"
+#define REDFISH_URI_LENGTH                      128
 
-  @retval EFI_SUCCESS              Service has been stoped successfully.
-  @retval EFI_INVALID_PARAMETER    This is NULL.
-  @retval Others                   Some error happened.
+#ifndef IS_EMPTY_STRING

```

Comment:
This is just suggestion. 

Looks like we have this definition in multiple files. We probably can have a header file like RedfishPkg/Include/RedfishBase.h to declare this useful macro.

Reply:
Good idea, but should it be a part of this PR? or will we have a separate request later?

Reply:
Yes we can do this in separated pull request later.

Reply:
Thank you!